examples/stable_diffusion_xl/gm/modules/embedders/modules.py

```diff
@@ -27,7 +22,7 @@ def __init__(self):
         self._input_key = None
 
     def tokenize(self, x):
-        raise NotImplementedError
+        return x, None
```

Comment:
this is a abstract class, the tokenize function is a abstract method leaving for the inherited class. It is a conventional practise to use the "NotImplementedError" in the abstract class

Reply:
What if we completely remove the `tokenize` method from all embedders, except for text embedders? As the concept of tokenization doesn’t apply to images, videos, fps, etc. Then, in `gm.modules.embedders.modules.GeneralConditioner.tokenize`, we could add a condition:
```python
if not hasattr(embedder, "tokenize"):
    emb_token, emb_length = batch[embedder.input_key], None
```
It’s not an optimal solution, but it can be fixed later during the refactoring stage.

Reply:
I think the `tokenize` method is more like a preprocessing method. For timestep embedding, it just return itself. We can change the name to a better way later, without modifying the code strucuture like impose a conditional branch

Reply:
I agree, imposing a conditional branch unnecessary overcomplicates code. That's why I've set `tokenize` to simply return `x, None` as in the case of image, video, time, fps, motion, and other embedders, it has to return itself. Let's address this during the refactoring stage.

Reply:
yes same as below this default method can be easily missed when doing implementation. let the user handle this